pkg/transform/cluster_report_transform.go

```diff
@@ -51,6 +58,52 @@ func genClusterReport(apiResources api.Resources) (ClusterReport, error) {
 	return clusterReport, nil
 }
 
+// reportNodes fills in information about nodes
+func (clusterReport *ClusterReport) reportNodes(apiResources api.Resources) {
+	logrus.Debug("ClusterReport::ReportNodes")
+
+	for _, node := range apiResources.NodeList.Items {
+		nodeReport := &NodeReport{
+			Name: node.ObjectMeta.Name,
+		}
+
+		isMaster, ok := node.ObjectMeta.Labels["node-role.kubernetes.io/master"]
+		if ok && isMaster == "true" {
+			nodeReport.MasterNode = true
+		} else {
+			nodeReport.MasterNode = false
+		}
+
+		reportResources(nodeReport, node.Status, apiResources)
+		clusterReport.Nodes = append(clusterReport.Nodes, *nodeReport)
+	}
+}
+
+// reportResources parse and insert info about consumed resources
+func reportResources(repotedNode *NodeReport, nodeStatus k8sapicore.NodeStatus, apiResources api.Resources) {
+	repotedNode.Resources.CPU = nodeStatus.Capacity.Cpu().String()
+
+	repotedNode.Resources.MemoryCapacity = nodeStatus.Capacity.Memory().String()
+
+	memConsumed := new(resource.Quantity)
+	memCapacity, _ := nodeStatus.Capacity.Memory().AsInt64()
+	memAllocatable, _ := nodeStatus.Allocatable.Memory().AsInt64()
+	memConsumed.Set(memCapacity - memAllocatable)
+	memConsumed.Format = resource.BinarySI
+	repotedNode.Resources.MemoryConsumed = memConsumed.String()
+
+	runningPods := 0
+	for _, resources := range apiResources.NamespaceMap {
+		for _, pod := range resources.PodList.Items {
+			if pod.Spec.NodeName == repotedNode.Name {
+				runningPods++
+			}
+		}
+	}
+	repotedNode.Resources.RunningPods = strconv.Itoa(runningPods) + "/" + nodeStatus.Capacity.Pods().String()
```

Comment:
@Danil-Grigorev what is the reason to have the string field in structure when you convert int to string here? Isn't it better to have RunningPod defined as int in the structure? May be I'm missing the story of your inspiration here, if so - share!

Reply:
I'll just quote Jason on this topic:

'There was a statement made that in some places IT sets up a cluster and lets the devs at it. They start spinning things up and IT really has no idea what's where.'

'so the capacity stuff, routes exposed, is to try and give them a fuller view of what's happened since they set it up'

'how many pv's totalling how much (at least potential storage), how many nodes, how much of those nodes are being utilized, etc.'

Certainly, from the code perspective having explicit conversion from int to string is stupid.

But having a structure like `pod_count/max_pod_count` implies having some conversions in place.

In terms of the idea and a customer perspective this output looks clear. Report should contain human-readable values, that could be captured from the first glance.

Reply:
I would agree in case the json is meant to be edited by humans. in our case it is just a mean to deliver the information to another application. I would insist on keeping this typed.

However if Jason have a reason, besides IT/humans doesn't know what is int/string in json, I am all for the support.

IOW, this reporting json output aren't meant for direct user consumption, even if - it is read only.

@jmontleon wdyt?